Make the forwarder operators reply to /healthz (#49)
DataDog/datadog-operator: vendor/github.com/heptiolabs/healthcheck/README.md

# healthcheck
[![Build Status](https://travis-ci.org/heptiolabs/healthcheck.svg?branch=master)](https://travis-ci.org/heptiolabs/healthcheck)
[![Go Report Card](https://goreportcard.com/badge/github.com/heptiolabs/healthcheck)](https://goreportcard.com/report/github.com/heptiolabs/healthcheck)
[![GoDoc](https://godoc.org/github.com/heptiolabs/healthcheck?status.svg)](https://godoc.org/github.com/heptiolabs/healthcheck)

Healthcheck is a library for implementing Kubernetes [liveness and readiness](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/) probe handlers in your Go application.

## Features

>  - Integrates easily with Kubernetes. This library explicitly separates liveness vs. readiness checks instead of lumping everything into a single category of check.

Reviewer: ```suggestion
 - This library integrates easily with Kubernetes because it explicitly separates liveness vs. readiness checks instead of lumping everything into a single category of check.
```

Author: Hello @apigirl ,

This is “vendored” code.
I.e.: this is a 3rd party dependency that the `datadog-operator` depends on and that has been copied verbatim inside this repo in order to not depend on the external world.
Such vendored 3rd party component must remain unmodified compared to upstream.
In order to fix the wording of this documentation, a PR should be submitted upstream instead: https://github.com/heptiolabs/healthcheck

   apiVersion: v1
   kind: Pod
   metadata:
     name: heptio-healthcheck-example
   spec:
     containers:
     - name: liveness
       image: your-registry/your-container

       # define a liveness probe that checks every 5 seconds, starting after 5 seconds
       livenessProbe:
         httpGet:
           path: /live
           port: 8086
         initialDelaySeconds: 5
         periodSeconds: 5

       # define a readiness probe that checks every 5 seconds
       readinessProbe:
         httpGet:
           path: /ready
           port: 8086
         periodSeconds: 5
   ```

 - If one of your readiness checks fails, Kubernetes will stop routing traffic to that pod within a few seconds (depending on `periodSeconds` and other factors).

 - If one of your liveness checks fails or your app becomes totally unresponsive, Kubernetes will restart your container.

 ## HTTP Endpoints
 When you run `go http.ListenAndServe("0.0.0.0:8086", health)`, two HTTP endpoints are exposed:

  - **`/live`**: liveness endpoint (HTTP 200 if healthy, HTTP 503 if unhealthy)
  - **`/ready`**: readiness endpoint (HTTP 200 if healthy, HTTP 503 if unhealthy)

> Pass the `?full=1` query parameter to see the full check results as JSON. These are omitted by default for performance.

Reviewer: ```suggestion
Pass the `?full=1` query parameter to see the full check results as JSON. This parameter is excluded by default for performance.
```

Author: Vendored code as well.
